vagrant/provisioning/roles/solr-custom-plugin/tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+- name: create custom solr plugin library directory
+  become: yes
+  become_user: solr
+  file:
+    path: "{{ root_folder }}/app/solr/contrib/arkcase/lib/"
+    state: directory
+  when: "enable_solr_snippets | default(false)"
+
+- name: create public key for signing Solr plugin jar (jar sig must be provided)
+  become: yes
+  command: openssl rsa -in {{ solr_plugin_pem_file | default('/etc/tls/private/arkcase-root.pem') }} -pubout -outform DER -out {{ root_folder }}/common/solr_plugin_key.der
+  when: "(ark_solr_custom_jar_sig is defined and ark_solr_custom_jar_sig != "" and enable_solr_snippets)  | default(false)"
```

Comment:
Nitpick: you can define a variable e.g. `sign_jar` based on these three booleans, and in the remaining conditions just refer to `sign_jar`